examples/upgradable-examples/upgradable_base/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+use near_plugins::{Ownable, Upgradable};
+use near_sdk::near_bindgen;
+use borsh::{BorshSerialize, BorshDeserialize};
+
+#[near_bindgen]
+#[derive(Ownable, Upgradable, Default, BorshSerialize, BorshDeserialize)]
+#[cfg(feature = "counter1")]
+struct Counter1 {
+  counter: u64,
+}
+
+#[near_bindgen]
+#[cfg(feature = "counter1")]
+impl Counter1 {
+  /// Specify the owner of the contract in the constructor
+  #[init]
+  pub fn new() -> Self {
+      let mut contract = Self { counter: 0 };
+      contract.owner_set(Some(near_sdk::env::predecessor_account_id()));
+      contract
+  }
+
+  pub fn inc1(&mut self) {
+      self.counter += 1;
+  }
+
+  pub fn get_counter(&self) -> u64 {
+      self.counter
+  }
+}
+
+#[near_bindgen]
+#[derive(Ownable, Upgradable, Default, BorshSerialize, BorshDeserialize)]
+#[cfg(feature = "counter2")]
+struct Counter2 {
+    counter: u64,
+}
+
+#[near_bindgen]
+#[cfg(feature = "counter2")]
+impl Counter2 {
+    #[init]
+    pub fn new() -> Self {
+        let mut contract = Self { counter: 0 };
+        contract.owner_set(Some(near_sdk::env::predecessor_account_id()));
+        contract
+    }
+
+    pub fn inc2(&mut self) {
+        self.counter += 2;
+    }
+
+    pub fn get_counter(&self) -> u64 {
+        self.counter
+    }
+}
```

Comment:
There's no need to create 2 separate structs and implementations that are the same. The more clean approach would be to have one `Counter`  contract and hide `inc2()` and/or `inc1()` by the feature flag. Or maybe even having just one `inc()` method, which changes the incrementation value depending on the feature.

Reply:
Fixed: https://github.com/aurora-is-near/near-plugins/pull/34/commits/f5b3b05e9f8c9e00acc45a5ed6d119d1d8d3c553